types/index.d.ts

```diff
@@ -1,24 +1,27 @@
 import { Options } from './options';
+import { ChartDatasetProperties, ChartType, ChartTypeRegistry, DeepPartial, IChartType, IPlugin } from "chart.js";
 
 declare module 'chart.js' {
-	interface ChartDataSets {
+	interface ChartDatasetProperties<TYPE extends ChartType, DATA extends unknown[]> {
```

Comment:
Unfortunately:
`TS2428: All declarations of 'ChartDatasetProperties' must have identical type parameters.`

Reply:
That's weird, this error should only apply to the parameter types, not the names. 

Did you try this?

```ts
interface ChartDatasetProperties<T extends ChartType, D extends unknown[]> {
```

Reply:
Yes, that's what I have tried. I believe it's because we are trying to extend the type there, and if type parameters don't match, the TS parser won't be able to match them.

Reply:
@etimberg I'm not sure that's a common practice to write generic parameter full uppercase (whatever language actually, e.g. template in C++). I think uppercase are mostly understood as constants or defines. Do you know why this choice in Chart.js definition types?

Reply:
I think it was an extension from the usual single letters `ChartDatasetProperties<T, D>` so that it was clear what they were. I have no big preference on all caps or not. I looked at https://www.typescriptlang.org/docs/handbook/generics.html but it only uses single letter generics.

Reply:
I like the explicit names, but I think [`Type, Data`](https://github.com/vuejs/vue/blob/dev/types/options.d.ts#L67) or [`TType, TData`](https://github.com/microsoft/TypeScript/blob/master/lib/lib.es2015.generator.d.ts#L23) are more suitable. Uppercase names may also not be accepted once TS code will be linted in the main repo (it would expect CamelCase) and is confusing with [constants](https://github.com/facebook/docusaurus/blob/487a9f98e434b4e8bf1da75ab60cd5f7ba039840/packages/docusaurus/src/webpack/base.ts#L21).

Reply:
In chart.js types there are a few generic types with 4/5 parameters: I think single letter types would turn out to be a bit too cryptical for those.

Reply:
https://github.com/chartjs/Chart.js/issues/7655#issuecomment-729933237